### ISSUE TITLE
Namespace auto-switching breaks when switching namespaces during a long-running evaluation

### ISSUE THREAD

**Issue by eerohele** [eerohele/Tutkain#92]
Repro:

```clojure
;; src/foo/bar.clj
(ns foo.bar)

(Thread/sleep 5000)

;; src/baz/quux.clj
(ns baz.quux)

(defn f [x] x)

(f 1)
```

Eval the sleep first, then `(f 1)`, ns is `foo.bar` when it should be `baz.quux`.

Maybe rethink approach: in the REPL, instead of blocking on reader, block on concurrent queue instead waiting for eval context, etc.

**Comment by eerohele** [comment]
>Maybe rethink approach: in the REPL, instead of blocking on reader, block on concurrent queue instead waiting for eval context, etc.

This will break sending multiple forms at once, because the REPL will wait for an eval context after each form. Back to the drawing board.

**Comment by eerohele** [comment]
Fixed in v0.14.1.
